scripts/install_third_party.py

```diff
@@ -330,11 +332,38 @@ def download_manifest_files(filepath):
                     dependency_tar_root_name, dependency_target_root_name)
 
 
+def remove_emojis_from_guppy(guppy_file_path):
+    """The emoji symbols present in the guppy.min.js file are not compatible
+    with all platforms, hence they must be removed from the file. Leaving these
+    leads to console warnings which cause the e2e tests to fail.
+    For reference: https://github.com/daniel3735928559/guppy/issues/138
+
+    Args:
+        guppy_file_path: str. Path of the guppy.min.js file.
+    """
+
+    text_to_remove = (
+        u'utf8chars:{template:"utf8char",values:{banana:{char:"🍌"},pineapple:'
+        u'{char:"🍍"},mango:{char:"🥭"},kiwi:{char:"🥝"}}},')
+
+    with codecs.open(guppy_file_path, mode='r', encoding='utf-8') as f:
+        file_content = f.read()
+
+    file_content = file_content.replace(text_to_remove, u'')
+
+    with codecs.open(guppy_file_path, mode='w', encoding='utf-8') as f:
+        f.write(file_content)
+
+
 def main(args=None):
     """Installs all the third party libraries."""
     unused_parsed_args = _PARSER.parse_args(args=args)
     download_manifest_files(MANIFEST_FILE_PATH)
 
+    guppy_file_path = os.path.join(
+        THIRD_PARTY_STATIC_DIR, 'guppy-f944022', 'build', 'guppy.min.js')
+    remove_emojis_from_guppy(guppy_file_path)
```

Comment:
I'm not sure whether we should do this kinda changes through this script inside a third-party library. Should we use a forked repo (Which doesn't have this unwanted emojis) until the main repo resolves the issue? Do you see any such pattern in the codebase for any other third party libs?

/cc @seanlip any thoughts?

Reply:
I discussed this with @aks681 and we concluded that this is the most reasonable solution given the situation.

> Should we use a forked repo (Which doesn't have this unwanted emojis) until the main repo resolves the issue?

That's a good idea, but I'm not sure if this is considered as a major issue since only one other person has reported it till now, so it doesn't seem like this will be fixed anytime soon. The [original issue](https://github.com/daniel3735928559/guppy/issues/138) was filed more than a year ago and hasn't received any activity since.

> Do you see any such pattern in the codebase for any other third party libs?

No -- I know this seems kinda dangerous, but it's a simple attribute removal from the dict present in the lib's file. It isn't messing with the logic in any way, which is why I think this is safe enough.

Reply:
Can't you provide your own symbols file as mentioned by the other poster?

Reply:
I would also suggest making a PR to the original repo upstream.

Reply:
> Can't you provide your own symbols file as mentioned by the other poster? I agree with @DubeySandeep that this is not a pattern to be followed.

The problem isn't due to the symbols file. That's just a json -- having these characters there doesn't cause these warnings. The problem is that these characters are present in guppy.js file ([here](https://github.com/daniel3735928559/guppy/blob/f9440220a73685ea9011c285a4af798f381de53a/build/guppy.js#L1162)), which is being compiled and processed, thus causing these warnings.

> I would also suggest making a PR to the original repo upstream.

Sorry, I don't understand, PR for what? For removing these symbols? As I mentioned, I don't think this is considered as a major issue, since no activity has been observed on the issue thread.
Again, I fail to understand why this is a problem. This is simply removing a variable from the js file, and not messing with the logic at all.

Reply:
I was referring to [this comment](https://github.com/daniel3735928559/guppy/issues/138#issuecomment-476901715). The person mentions that they solved this issue that way. Is the issue you are running to the same one, and if so, could you do something similar to what they did?

I was also suggesting making a PR upstream to fix this issue. No activity on an issue doesn't mean it's not major. We have lots of issues here but not enough time to fix them, so you shouldn't take that as the sole signal.

Finally, we generally do not modify third-party libs on the fly like this. This is because it results in different contributors ending up with different versions of the libs and tends to break our versioning system. In general we prefer contributing patches upstream.

Reply:
> I was referring to this comment. The person mentions that they solved this issue that way. Is the issue you are running to the same one, and if so, could you do something similar to what they did?

Yes, I realize that, but I don't think it's the exact same problem. I've tested this locally and modifying the symbols file still gave the same warning. This is probably because of the fact that these characters have been hard-coded in the guppy.js file. They probably wouldn't have been in the js file at the time of filing this issue which is why having a custom symbols file might've solved the issue then, but the same solution doesn't work now.

> I was also suggesting making a PR upstream to fix this issue. No activity on an issue doesn't mean it's not major. We have lots of issues here but not enough time to fix them, so you shouldn't take that as the sole signal.

I've created a [PR](https://github.com/daniel3735928559/guppy/pull/173) and commented on the issue thread. But will I have to wait for that PR to get merged before getting this one in? This PR blocks further progress in the project.